models/challenge_model.go

```diff
@@ -14,21 +14,32 @@ type Location struct {
 	Coordinates []float64 `json:"coordinates" bson:"coordinates" binding:"required"`
 }
 
+// Location (lat/long) and Address information are optional.
+// Case 1: Location (provided), Address (provided)
+//      - Create object as usual.
+// Case 2: Location (provided), Address (not provided)
+//      - Create object with Location, but no Address.
+// Case 3: Location (not provided), Address (provided)
+//      - Geocode address and retrieve lat/long, create
+//        object with Location & Address.
+// Case 4: Location (not provided), Address (not provided)
+//      - Create object as usual, without both information.
+//        Validate duplicates based only on taskName.
```

Comment:
question: why do we support case 4? in general, do we allow challenges that don't have a particular "place" to do it (maybe for example, play a game of scribbl something)?

Reply:
yep -- I think we decided a while back there can be non-location-based challenges that we offer.